### PR TITLE
initial commit for python backend

### PR DESCRIPTION
**Pull request by chernyee**

### CODE REVIEW

**Review by NadeemYaseen — APPROVED** [review]

File: backend/submodule/clock.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        total_power = 0
+        for clock in self.clocks:
+            total_power += clock.compute_dynamic_power()
+        return total_power
+
+if __name__ == '__main__':
+    # manual test code
+    clock = Clock(enable=True)
+    clock2 = Clock(enable=True)
+    print(clock)
```
Comment (ravikiranchollangi) [inline backend/submodule/clock.py:135]: Please replace with the logger class in future PR. We don't want print statements on the console.

File: tests/test_rpe.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#  Copyright (C) 2024 RapidSilicon
+#  Authorized use only
+#
+import pytest
+# from backend.rpe import RPE
+
+
+def test_rpe_1():
+    assert 1 == 1
+    # rpe = RPE('/home/guanyung/dev/rpe-poc/python/etc/device.xml')
```
Comment (ravikiranchollangi) [inline tests/test_rpe.py:11]: Please add TODOs in code instead of commenting code when possible.

File: backend/submodule/clock.py (hunk 1 above)
Comment (chernyee) [reply backend/submodule/clock.py:135]: Noted.

File: backend/etc/device.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<device_list>
```
Comment (ravikiranchollangi) [inline backend/etc/device.xml:1]: For now, this is ok but we should be getting it from raptor repo as discussed.

**Review by ravikiranchollangi — APPROVED** [review]

**Comment by ravikiranchollangi** [comment]
Please also add "README" in separate PR.

File: backend/etc/device.xml (hunk 3 above)
Comment (chernyee) [reply backend/etc/device.xml:1]: noted.

File: tests/test_rpe.py (hunk 2 above)
Comment (chernyee) [reply tests/test_rpe.py:11]: fixed

**Pull request merged**